### PR TITLE
Bazefetcher source - roots list + docs

### LINKED ISSUE #88: Bazefetcher source should be able to take a list of root directories
requires #60

### LINKED ISSUE #94: docstring: bazefetcher source
child of #73

### PR DESCRIPTION
**Pull request by achaikou**
fixes #88 
fixes #94

### CODE REVIEW

File: camille/source/bazefetcher.py [hunk 1]
```diff
@@ -98,10 +133,11 @@ def _extend_fwd(end_date, df, tag_root, fn_regex, tzinfo):
...
         return df, end_date + datetime.timedelta(microseconds=1)
 
-    files = [
-        os.path.join(tag_root, fn) for fn in os.listdir(tag_root)
-        if fn_regex.match(fn) and _fn_start_date(fn) > end_date
-    ]
+    def date_func(fn):
+        return _fn_start_date(fn) > end_date
+
+    files = _get_files(src_dirs, tag, fn_regex, date_func)
```
Comment (JensGM) [inline camille/source/bazefetcher.py]: suggestion:
```suggestion
    files = _get_files(src_dirs, tag, fn_regex, lambda fn: _fn_start_date(fn) > end_date)
```

File: camille/source/bazefetcher.py [hunk 2]
```diff
@@ -125,9 +161,56 @@ def _extend_fwd(end_date, df, tag_root, fn_regex, tzinfo):
...
+           Result time series
+
+    Examples
+    --------
+
+    Read time series `series` with tag `tag`:
+
+    >>> start_date = datetime.datetime(2029, 1, 1, tzinfo=pytz.utc)
+    >>> end_date = datetime.datetime(2030, 1, 1, tzinfo=pytz.utc)
+    >>> cin = camille.input.bazefetcher('<root-directory>')
```
Comment (JensGM) [inline camille/source/bazefetcher.py]: More examples!

File: camille/source/bazefetcher.py [hunk 3]
```diff
@@ -125,9 +161,56 @@ def _extend_fwd(end_date, df, tag_root, fn_regex, tzinfo):
...
+    >>> ts = cin('tag', start_date, end_date)
+    """
+    if isinstance(src_dir, str):
+        src_dir = [src_dir]
+
+    src_dirs = [dir for dir in src_dir if os.path.isdir(dir)]
```
Comment (JensGM) [inline camille/source/bazefetcher.py]: dir is buildin
```suggestion
    src_dirs = [d for d in src_dir if os.path.isdir(d)]
```
```suggestion
    src_dirs = list(map(os.path.isdir, src_dir))
```

File: camille/source/bazefetcher.py [hunk 4]
```diff
@@ -53,7 +53,41 @@ def _tidy_frame(df, tzinfo):
     df.sort_index(inplace=True)
 
 
-def _extend_bwd(start_date, df, tag_root, fn_regex, tzinfo):
+def _get_files(src_dirs, tag, fn_regex, date_func):
```
Comment (JensGM) [inline camille/source/bazefetcher.py]:
```suggestion
def _get_files(src_dirs, tag, fn_regex, date_pred):
```

File: camille/source/bazefetcher.py [hunk 5]
```diff
@@ -53,7 +53,41 @@ def _tidy_frame(df, tzinfo):
...
+
+        files = [] if files is None else files
+        files = files + [
+            os.path.join(tag_root, fn) for fn in os.listdir(tag_root)
+            if fn_regex.match(fn) and date_func(fn)
+        ]
+
+    if files is None:
+        raise ValueError('Tag {} not found in {}'.format(tag, src_dirs))
+
```
Comment (JensGM) [inline camille/source/bazefetcher.py]:
```python
    tag_roots = list(filter(isdir, (join(dr, tag) for dr in src_dirs)))

    if not tag_roots:
        raise ValueError('Tag {} not found in {}'.format(tag, src_dirs))
    
    files = [join(r, fn) for fn in os.listdir(r) for r in tag_roots
             if fn_regex.match(fn) and date_func(fn)]
```

File: tests/source/test_bazefetcher.py [hunk 6]
```diff
@@ -164,3 +164,39 @@ def test_snap_both():
...
+
+    sin_b = baze_and_authored('Sin-T60s-SR01hz', t1_2, t1_4)
+    assert len(sin_b) == 17280
+
+    i04_status_b = authored('installation-04-status', t12_31_23, t1_5_1)
+    assert len(i04_status_b) == 5
+
+
+def test_many_roots_same_tag():
+    roots = ['tests/test_data/authored/many_roots/dir'+ str(index)
```
Comment (JensGM) [inline tests/source/test_bazefetcher.py]: Move many_roots up a level

File: camille/source/bazefetcher.py [hunk 7]
```diff
@@ -142,20 +225,13 @@ def bazefetcher_internal(tag,
...
         fn_regex = re.compile(tag + fn_tail_pattern)
 
-        files = [
-            os.path.join(tag_root, fn) for fn in os.listdir(tag_root)
-            if fn_regex.match(fn)
-                and _fn_start_date(fn) <= end_date
-                and start_date <= _fn_end_date(fn)
-        ]
-        files.sort()
+        def date_func(fn):
```
Comment (JensGM) [inline camille/source/bazefetcher.py]: lambda

**Comment by achaikou** [comment]
Done all discussed

**Pull request merged**